core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchReadableSource.java

```diff
@@ -37,18 +39,28 @@
  * @param <OUT> the type of output for the
  */
 public abstract class BatchReadableSource<KEY_IN, VAL_IN, OUT> extends BatchSource<KEY_IN, VAL_IN, OUT> {
+  private final BatchReadableWritableConfig batchReadableWritableConfig;
+
+  protected BatchReadableSource(BatchReadableWritableConfig batchReadableWritableConfig) {
+    this.batchReadableWritableConfig = batchReadableWritableConfig;
+  }
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
-    String name = getProperties().get(Properties.BatchReadableWritable.NAME);
-    Preconditions.checkArgument(name != null && !name.isEmpty(), "Dataset name must be given.");
-    String type = getProperties().get(Properties.BatchReadableWritable.TYPE);
-    Preconditions.checkArgument(type != null && !type.isEmpty(), "Dataset type must be given.");
+    // null check for tests, as macro fields map and properties are null in test cases
```

Comment:
Was left over from when there was a null check for `getProperties()`, forgot to remove it.